### DISCUSSION TITLE
How to use the halo mass range  galactic filter?

### DISCUSSION THREAD

**Discussion by liempi** [galacticusorg/galacticus#401] [Q&A] [answered]
Hi all :) 

I tried to use the haloMassRange galactic filter but is not working. Specifically, Galacticus returns:
`Fatal error:
[virialDensityContrastDefinition] object is undefined
 Occurred at:
    directive:objectBuilder
     function:haloMassRangeConstructorParameters
         file:galactic.filters.halo_mass_range.F90   [line 90]
  => Error occurred in master thread
`  
I attach an minimalist example and the full log file for the parameter file. Also, I'm using the last version of Galacticus. Anyone knows what could be happening? The problem occurs only when I use this filter.

Cheers 

[quickTest_filter.txt](https://github.com/galacticusorg/galacticus/files/11020869/quickTest_filter.txt)
[output.log](https://github.com/galacticusorg/galacticus/files/11020882/output.log)

**Comment by abensonca** [comment, marked as answer]
For this filter you need to tell it how you're defining halo masses. So, for example, if the halo masses you're using follow the M_200c convention (i.e. they are defined as the mass inside a sphere containing a mean density of 200 times the critical density), then you would do:
```
     <galacticFilter value ='haloMassRange'>
      <massLow  value ='1.0e4' />
      <massHigh value ='1.0e13'/>
      <virialDensityContrastDefinition value="fixed">
        <densityType value="fixed"/>
        <densityContrastValue value="200.0"/>
      </virialDensityContrastDefinition>
    </galacticFilter>
```
Galacticus will then convert your masses to the convention it uses internally. In your parameter file that internal convention is specified via this parameter:
```
  <virialDensityContrast value="sphericalCollapseClsnlssMttrCsmlgclCnstnt"/>
```

If the masses you're specifying to this filter follow this internal definition, then just use:
```
     <galacticFilter value ='haloMassRange'>
      <massLow  value ='1.0e4' />
      <massHigh value ='1.0e13'/>
       <virialDensityContrastDefinition value="sphericalCollapseClsnlssMttrCsmlgclCnstnt"/>
    </galacticFilter>
```

**Reply by liempi** [reply]
Ok, perfect. Many thanks!
